src/main/java/application/controller/CustomerController.java

```diff
@@ -0,0 +1,338 @@
+        	if (customerId == null) {
+        		// if no user passed in, this is a bad request
+        		return ResponseEntity.badRequest().body("Invalid Bearer Token: Missing customer ID");
+        	}
+        	
+        	logger.info("caller: " + customerId);
+			if (!customerId.equals("id")) {
+        		// if i'm getting a customer ID that doesn't match my own ID, then return 401
+        		return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        	}
+
+            final Database cloudant = getCloudantDatabase();
+            final Customer cust = getCloudantDatabase().find(Customer.class, id);
+    
+            cust.setFirstName(payload.getFirstName());
+            cust.setLastName(payload.getLastName());
+            cust.setImageUrl(payload.getImageUrl());
+            cust.setEmail(payload.getEmail());
+            
+            // TODO: hash password
+            cust.setPassword(payload.getPassword());
+            
+            cloudant.save(payload);
+        } catch (NoDocumentException e) {
+            logger.error("Customer not found: " + id);
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Customer with ID " + id + " not found");
+        } catch (Exception ex) {
+            logger.error("Error updating customer: " + ex);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error updating customer: " + ex.toString());
+        }
+        
+        return ResponseEntity.ok().build();
+    }
+
+    /**
+     * Delete customer 
+     * @return transaction status
+     */
+    @RequestMapping(value = "/customer/{id}", method = RequestMethod.DELETE)
+    protected ResponseEntity<?> delete(@RequestHeader Map<String, String> headers, @PathVariable String id) {
+		// TODO: no one should have access to do this, it's not exposed to APIC
+    	
+        try {
+            final Database cloudant = getCloudantDatabase();
+            final Customer cust = getCloudantDatabase().find(Customer.class, id);
+            
+
+            cloudant.remove(cust);
+        } catch (NoDocumentException e) {
+            logger.error("Customer not found: " + id);
+            return ResponseEntity.status(HttpStatus.NOT_FOUND).body("Customer with ID " + id + " not found");
+        } catch (Exception ex) {
+            logger.error("Error deleting customer: " + ex);
+            return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body("Error deleting customer: " + ex.toString());
+        }
+        return ResponseEntity.ok().build();
+    }
+
+    @SuppressWarnings("unused")
+	private ResponseEntity<?> failGetCustomers(@RequestHeader Map<String, String> headers) {
+        // Simply return an empty array
+        final List<Customer> inventoryList = new ArrayList<Customer>();
+        return ResponseEntity.ok(inventoryList);
+    }
+
+    /**
+     * @return Circuit breaker tripped
+     */
+    @HystrixCommand(fallbackMethod="failGood")
```

Comment:
For now circuit breaker is not implemented. We can later adopt it if required. For now we can remove the hystrix part and all spring cloud related dependencies.